lib/spack/spack/hooks/dashboards.py

```diff
@@ -0,0 +1,576 @@
+        value = ET.SubElement(completion_status, 'Value')
+        value.text = 'Completed'
+        # Command line
+        cmd_line = ET.SubElement(self.result, 'NamedMeasurement')
+        cmd_line.set('type', 'text/string')
+        cmd_line.set('name', 'Command Line')
+        value = ET.SubElement(cmd_line, 'Value')
+        value.text = "spack install"
+        # Logs
+        if message is not None:
+            logs = ET.SubElement(self.result, 'Measurement')
+            value = ET.SubElement(logs, 'Value')
+            value.text = message
+
+
+
+
+class CDashSimpleTestSuite(object):
+    def __init__(self, spec, filename, slot='Experimental'):
+        self.spec = spec
+        self.slot = slot
+        self.tests = []
+        self.buildstamp = "%s-%s" % (time.strftime("%Y%d%m-%H:%M:%S"), slot)
+        self.filename = filename
+
+    def create_filename(self, spec, subdir, step):
+        if "build" in str(step):
+            if self.filename is not None:
+                return "%s.%s.xml" % (self.filename, step)
+            else:
+                fmt = '%s-{x.name}-{x.version}-{hash}.xml' % step
+                basename = fmt.format(x=spec, hash=spec.dag_hash())
+                dirname = fs.join_path(spack.var_path, subdir)
+                fs.mkdirp(dirname)
+                return fs.join_path(dirname, basename)
+
+    def create_template(self):
+        template = ET.Element('Site')
+        buildName = str(self.spec.short_spec)
+        buildName = buildName.split('=')
+        template.set('BuildName',str(buildName[0]) + " " + str(buildName[1]))
+        template.set('BuildStamp', self.buildstamp)
+        tempcompiler = self.spec.short_spec.split('%')[1].split(' ')[0]
+        template.set('CompilerName', tempcompiler.split('@')[0])
+        template.set('CompilerVersion', tempcompiler.split('@')[1])
+        template.set('Name', platform.node())
+        template.set('OSName', "Mac OS X")
+        template.set('Hostname', platform.node())
+        #template.set('Type', self.slot)
+        return template
+
+    def create_testcase(self, name, spec):
+        item = CDashTestCase(name, spec)
+        self.tests.append(item)
+        return item
+
+    def dump(self):
+        build_report = self.prepare_build_report()
+        filename = self.create_filename(self.spec, 'cdash', 'build')
+        self.dump_report(build_report, filename)
+        
+
+    def dump_report(self, report, filename):
+        with codecs.open(filename, 'wb', 'utf-8') as file:
+            xml_string = ET.tostring(report)
+            xml_string = xml.dom.minidom.parseString(xml_string).toprettyxml()
+            file.write(xml_string)
+
+    def now(self):
+        #return time.strftime("%a %b %d %H:%M:%S %Z")
+        return time.strftime("%b %d %H:%M %Z")
+
+    def epoch(self):
+        return str(calendar.timegm(time.gmtime()))
+
+
+    def prepare_build_report(self):
+        report = self.create_template()
```

Comment:
I'm curious whether the XML parts of this would be more maintainable if you just make an XML template with `"""`, e.g.:

```
template = """\
<Build>
    <StartDateTime>{date}</StartDateTime>
    <StartDateTime>{log_text}</StartDateTime>
</Build>
"""
output = template.format(date=date, log_text=captured_output)
```

etc.  You can get pretty far with Python's `str.format()` function.

What do you think?  I think manually constructing the `ElementTree` is kind of tedious and hard to read or change.